pkg/sql/catalog/hydrateddesc/hydrate.go

```diff
@@ -82,6 +85,12 @@ func MakeTypeLookupFuncForHydration(
 		case catalog.TypeDescriptor:
 			typ = t
 		case catalog.TableDescriptor:
+			if !t.Hydrated() {
+				if err := Hydrate(ctx, t, typeLookupFunc); err != nil {
```

Comment:
`Hydrate` is not thread-safe, should we be concerned?

Reply:
Ignoring my concerns about the `skipHydration` boolean argument you probably should retrieve a hydrated descriptor using `lookupFn(ctx, id, false /* skipHydration */)` rather than calling `Hydrate` directly.

Reply:
I'm afraid that we need to do `Hydrate` because [this path](https://github.com/cockroachdb/cockroach/blob/cf1e7e6bc6ef9e55510b9ed13bd068bb3894cd92/pkg/sql/catalog/descs/hydrate.go#L169-L169) (used by show backup) really doesn't lookup anything, I'm assuming they have all descriptors in the fast lookup catalog....

Reply:
> Hydrate is not thread-safe, should we be concerned?

Hmm...I'll need to think about this.

Reply:
Where does multi-threading come into the picture here?

Reply:
I think we're fine here, The only thing shared across threads regarding this hydration domain is the hydrated descriptor cache and that is thread safe.

Reply:
That's a relief!